### PR TITLE
Adjust documentation, import paths and codegen to avoid case-insensitive import collision

### PR DESCRIPTION
**Pull request by mrow4a**
When following the steps from https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/docs/quick-start-guide.md#build and calling `go generate` and then `go test ./...` there were errors as 

`
../../googlecloudplatform/spark-on-k8s-operator/vendor/k8s.io/apimachinery/pkg/watch/until.go:23:2: case-insensitive import collision: "github.com/googlecloudplatform/spark-on-k8s-operator/vendor/k8s.io/apimachinery/pkg/util/wait" and "github.com/GoogleCloudPlatform/spark-on-k8s-operator/vendor/k8s.io/apimachinery/pkg/util/wait"
`

CC @liyinan926

### CODE REVIEW

File: docs/quick-start-guide.md [hunk 1]
```diff
@@ -214,8 +214,8 @@ $ docker build --build-arg SPARK_IMAGE=<your Spark image> -t <image-tag> .
 If you'd like to build/test the spark-operator locally, follow the instructions below:
 
 ```bash
-$ mkdir -p $GOPATH/src/k8s.io
-$ cd $GOPATH/src/k8s.io
+$ mkdir -p $GOPATH/src/github.com/googlecloudplatform
+$ cd $GOPATH/src/github.com/googlecloudplatform
 $ git clone git@github.com:GoogleCloudPlatform/spark-on-k8s-operator.git
```
Comment (mrow4a) [inline docs/quick-start-guide.md]: Here is adjusted and correct doc

Comment (mrow4a) [reply docs/quick-start-guide.md]: the `dep ensure` will not work, only `dep ensure -vendor-only` works

File: Gopkg.lock [hunk 2]
```diff
@@ -2,70 +2,30 @@
...
+    "autorest/date"
   ]
-  pruneopts = ""
   revision = "fc3b03a2d2d1f43fad3007038bd16f044f870722"
   version = "v9.10.0"
 
 [[projects]]
-  branch = "master"
-  digest = "1:a01ad406efc79e2cdf827baa8836eb7a0ceb84bcec7f389a163076748076d4ab"
-  name = "github.com/GoogleCloudPlatform/spark-on-k8s-operator"
```
Comment (mrow4a) [inline Gopkg.lock]: Now this does not get placed in Gopkg.lock

**Comment by liyinan926** [comment]
This is weird. The correct import paths should start with `github.com/GoogleCloudPlatform/spark-on-k8s-operator/...` instead of the lower-case one. For some reason, `go generate` generates the lower-case one.

**Comment by mrow4a** [comment]
@liyinan926 if you find the better way of solving this, just close this PR with better option. But I am afraid lowercasing all imports might be necessary

**Comment by mrow4a** [comment]
Hmm, I start actually thinking - will it actually not break the imports ? https://github.com/DAddYE/igo/issues/2

**Comment by liyinan926** [comment]
I guess it's more important to be consistent than to favor a particular case.

**Comment by liyinan926** [comment]
Let me dig a bit deeper on this.

**Comment by mrow4a** [comment]
@liyinan926 yes, please go ahead - I think lowercasing as in this PR wont solve the issue

**Pull request closed without merging**